unlock-app/src/components/helpers/ErrorMessage.js

```diff
@@ -14,7 +14,7 @@ const genericError = message => (
 const defaultMessages = {
   MISSING_PROVIDER: <p>You need a web3 provider to use Unlock.</p>,
   NOT_ENABLED_IN_PROVIDER: <p>Please enable Unlock in your web3 Provider</p>,
-  NON_DEPLOYED_CONTRACT: (
+  FATAL_NON_DEPLOYED_CONTRACT: (
```

Comment:
This introduces an inconsistency. `FATAL_` errors are intended to only be handled by the `GlobalErrorConsumer` and generated by the `GlobalErrorProvider` at startup, not by redux. Redux-based errors are handled by components and converted from error codes to static markup here.

This is also true of a recent change to the `FATAL_NO_USER_ACCOUNT` constant made in a separate PR.

Reply:
We are changing the way the errors are handled by getting rid of the provider and just using redux to dispatch the errors.
The fatal one will be rendered and block the app.